### DISCUSSION TITLE
Review OpenSRP CHW & Reveal Tasking pieces

### DISCUSSION THREAD

**Discussion by dubdabasoduba** [opensrp/fhircore#975] [Ideas]
**Pre-req**
- [ ] Understand Familly tasking vs individual tasking
- [ ] Understand the task workflow on reveal and CHW apps

**SubTask**
- [ ] #857 
- [ ] Current POC implementation - #844

**Comment by f-odhiambo** [comment]
@dubdabasoduba  Adding here some reference links for this

1. Scoping Docs - https://smartregister.atlassian.net/wiki/spaces/Documentation/pages/1156219033/Scoping+Docs 
2. FHIR Task-Based Referrals Software Requirements Specification - https://smartregister.atlassian.net/wiki/spaces/Documentation/pages/1156251738/FHIR+Task+Based+Referrals+Software+Requirements+Specification
3. Dynamic Tasking Sample Plan Definitions - https://smartregister.atlassian.net/wiki/spaces/Documentation/pages/1985576965/Dynamic+Tasking+Sample+Plan+Definitions
4. Task in FHIR  - https://www.hl7.org/fhir/task.html
5. Task - Status  - https://www.hl7.org/fhir/valueset-task-status.html#expansion. Currently using Requested, Ready & Complete for this ticket 
We will look for the current sample APK to also review the workflow as implemented e.g  Reveal

**Comment by f-odhiambo** [comment]
[OpenSRP Tasks status ](http://build.fhir.org/ig/HL7/sdc/extraction.html)

![image](https://user-images.githubusercontent.com/4540684/149733240-4351140b-a54a-406d-9db7-ea4124dad942.png)
**Task Statuses**

1. Draft - The task is not yet ready to be acted upon.
2. Ready - Task is ready to be performed, but no action has yet been taken. Used in place of requested/received/accepted/rejected when request assignment and acceptance is a given.
3. In-Progress - Task has been started but is not yet complete.
4. Cancelled - The task was not completed.
5. Completed - The task has been completed.
6. Failed - The task was attempted but could not be completed due to some error.
7. Archived - The task is invalid, a duplicate or is no longer required for reporting purposes.

**Task Business Statuses**
1. Referred - The referral was sent
2. In-Progress - The referral was opened by the facility worker
3. Complete - The task was marked as done
4. Expired - The task expired before it was done


**FHIR Task Status**



Code | Display | Definition | Canonical Status
-- | -- | -- | --
draft | Draft | The task is not yet ready to be acted upon. | ~draft
**requested** | Requested | The task is ready to be acted upon and action is sought. | ~requested
received | Received | A potential performer has claimed ownership of the task and is evaluating whether to perform it. | ~received
accepted | Accepted | The potential performer has agreed to execute the task but has not yet started work. | ~accepted
rejected | Rejected | The potential performer who claimed ownership of the task has decided not to execute it prior to performing any action. | ~declined
**ready** | Ready | The task is ready to be performed, but no action has yet been taken. Used in place of requested/received/accepted/rejected when request assignment and acceptance is a given. | ~on-target
cancelled | Cancelled | The task was not completed. | ~abandoned
in-progress | In Progress | The task has been started but is not yet complete. | ~active
on-hold | On Hold | The task has been started but work has been paused. | ~suspended
failed | Failed | The task was attempted but could not be completed due to some error. | ~failed
**completed** | Completed | The task has been completed. | ~complete
entered-in-error | Entered in Error | The task should never have existed and is retained only because of the possibility it may have used. | ~error

**Comment by f-odhiambo** [comment]
Currently using Requested, Ready & Complete for this ticket -#844

**Comment by f-odhiambo** [comment]
Test data - https://turn-fhir.smartregister.org/fhir/Patient/6741/$everything

**Comment by f-odhiambo** [comment]
@dubdabasoduba Converting this to a discussion item
